persistence/src/androidMain/kotlin/com/wire/kalium/persistence/kmm_settings/EncryptedSettingsHolder.kt

```diff
@@ -2,20 +2,25 @@ package com.wire.kalium.persistence.kmm_settings
 
 import android.content.Context
 import androidx.security.crypto.EncryptedSharedPreferences
-import androidx.security.crypto.MasterKeys
+import androidx.security.crypto.MasterKey
 import com.russhwolf.settings.AndroidSettings
 import com.russhwolf.settings.Settings
 
 actual class EncryptedSettingsHolder(
-    applicationContext: Context,
-    options: SettingOptions,
-    masterKeyAlias: String = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)
+    private val applicationContext: Context,
+    options: SettingOptions
 ) {
+    private fun getOrCreateMasterKey(): MasterKey = MasterKey
+        .Builder(applicationContext)
+        .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+        .setRequestStrongBoxBacked(true)
```

Comment:
Does this work in cases where no StrongBox backend is available?
I would expect this to fail with an exception.

Reply:
i did try is on emulater that doesn't have StrongBox support and it created keystoreBackedKey

also did check the `MasterKey.java` class and they do the check if strongBox is avilable 

```kalium
        public MasterKey build() throws GeneralSecurityException, IOException {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                return buildOnM();
            } else {
                return new MasterKey(mKeyAlias, null);
            }
        }

 private MasterKey buildOnM() throws GeneralSecurityException, IOException {
--------------------
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && mRequestStrongBoxBacked) {
                    if (mContext.getPackageManager().hasSystemFeature(
                            PackageManager.FEATURE_STRONGBOX_KEYSTORE)) {
                        builder.setIsStrongBoxBacked(true);
                    }
                }
```

I can check for `GeneralSecurityException` but not sure how to handle such exceptions

Reply:
The documentation was not entirely clear to me. Now with the referenced code snippet, I'm confident, that there is no change needed and the Builder takes care of what I worried about.